examples/egl_common.c

```diff
@@ -0,0 +1,168 @@
+		const EGLint *config_attribs) {
+	struct egl_ctx *egl = calloc(1, 
+		sizeof(struct egl_ctx));
+	if (egl == NULL) {
+		wlr_log(WLR_ERROR, "Allocation failed");
+		return NULL;
+	}
+
+	const char *client_exts_str = 
+		eglQueryString(EGL_NO_DISPLAY, 
+			EGL_EXTENSIONS);
+	if (client_exts_str == NULL) {
+		if (eglGetError() == EGL_BAD_DISPLAY) {
+			wlr_log(WLR_ERROR, 
+				"EGL_EXT_client_extensions not supported");
+		} else {
+			wlr_log(WLR_ERROR, 
+				"Failed to query EGL client extensions");
+		}
+		return NULL;
+	}
+
+	if (!strstr(client_exts_str, 
+			"EGL_EXT_platform_base")) {
+		wlr_log(WLR_ERROR, 
+			"EGL_EXT_platform_base not supported");
+		return NULL;
+	}
+
+	if (!strstr(client_exts_str, 
+			"EGL_EXT_platform_wayland")) {
+		wlr_log(WLR_ERROR, 
+			"EGL_EXT_platform_wayland not supported");
+		return NULL;
+	}
+
+	egl->eglGetPlatformDisplayEXT = 
+		(void *)eglGetProcAddress("eglGetPlatformDisplayEXT");
+	if (egl->eglGetPlatformDisplayEXT == NULL) {
+		wlr_log(WLR_ERROR, 
+			"Failed to get eglGetPlatformDisplayEXT");
+		return NULL;
+	}
+
+	egl->eglCreatePlatformWindowSurfaceEXT = 
+		(void *)eglGetProcAddress("eglCreatePlatformWindowSurfaceEXT");
+	if (egl->eglCreatePlatformWindowSurfaceEXT == NULL) {
+		wlr_log(WLR_ERROR, 
+			"Failed to get eglCreatePlatformWindowSurfaceEXT");
+		return NULL;
+	}
+
+	egl->display = 
+		egl->eglGetPlatformDisplayEXT(EGL_PLATFORM_WAYLAND_EXT, 
+			display, NULL);
+	if (egl->display == EGL_NO_DISPLAY) {
+		wlr_log(WLR_ERROR, "Failed to create EGL display");
+		goto error;
+	}
+
+	EGLint major, minor;
+	if (eglInitialize(egl->display, &major, 
+			&minor) == EGL_FALSE) {
+		wlr_log(WLR_ERROR, "Failed to initialize EGL");
+		goto error;
+	}
+
+	EGLint matched = 0;
+	if (!eglChooseConfig(egl->display, config_attribs, 
+			&egl->config, 1, &matched)) {
+		wlr_log(WLR_ERROR, "eglChooseConfig failed");
+		goto error;
+	}
+	if (matched == 0) {
+		wlr_log(WLR_ERROR, "Failed to match an EGL config");
+		goto error;
+	}
+
+	size_t atti = 0;
+	EGLint attribs[5];
+	attribs[atti++] = 
+		EGL_CONTEXT_CLIENT_VERSION;
+	attribs[atti++] = 2;
+
+	attribs[atti++] = 
+		EGL_NONE;
+	assert(atti <= 
+		sizeof(attribs)/sizeof(attribs[0]));
+
+	egl->context = 
+		eglCreateContext(egl->display, egl->config, 
+			EGL_NO_CONTEXT, attribs);
+	if (egl->context == EGL_NO_CONTEXT) {
+		wlr_log(WLR_ERROR, "Failed to create EGL context");
+		goto error;
+	}
+
+	return egl;
+
+error:
+	eglMakeCurrent(EGL_NO_DISPLAY, EGL_NO_SURFACE, 
+		EGL_NO_SURFACE, EGL_NO_CONTEXT);
+	if (egl->display) {
+		eglTerminate(egl->display);
+	}
+	eglReleaseThread();
+	return NULL;
+}
+
+void egl_context_destroy(struct egl_ctx *egl) {
+	if (egl == NULL) {
+		return;
+	}
+
+	eglMakeCurrent(egl->display, EGL_NO_SURFACE, 
+		EGL_NO_SURFACE, EGL_NO_CONTEXT);
+	eglDestroyContext(egl->display, egl->context);
+	eglTerminate(egl->display);
+	eglReleaseThread();
+
+	free(egl);
+}
+
+EGLSurface create_egl_surface(struct egl_ctx *egl, void *window) {
```

Comment:
I'd just call `eglCreatePlatformWindowSurfaceEXT` from the examples directly instead of having this extra wrapper.

Reply:
Done!